From c2232d5075342347a8ff814ced5ce8b9e1cf64b9 Mon Sep 17 00:00:00 2001
From: Thomas Leonard <talex5@gmail.com>
Date: Wed, 7 Apr 2010 15:44:57 +0100
Subject: [PATCH] Bugfix: unselect item when menu is closed

We used to listen for "unmap_event", but this is no longer emitted in recent
versions of GTK. Switched to using "selection-done" instead.

Closes #2925212 (reported by Barry Kauler).
---
 ROX-Filer/src/menu.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ROX-Filer/src/menu.c b/ROX-Filer/src/menu.c
index 611918d..bf4bae3 100644
--- a/ROX-Filer/src/menu.c
+++ b/ROX-Filer/src/menu.c
@@ -322,9 +322,9 @@ gboolean ensure_filer_menu(void)
 	GET_SSMENU_ITEM(item, "filer", "Window", "New Window");
 	filer_new_window = GTK_BIN(item)->child;
 
-	g_signal_connect(filer_menu, "unmap_event",
+	g_signal_connect(filer_menu, "selection-done",
 			G_CALLBACK(menu_closed), NULL);
-	g_signal_connect(filer_file_menu, "unmap_event",
+	g_signal_connect(filer_file_menu, "selection-done",
 			G_CALLBACK(menu_closed), NULL);
 
 	g_signal_connect(filer_keys, "accel_changed",
-- 
1.6.5.GIT

